runbotics/runbotics-desktop/src/action/jira/jira.utils.ts

```diff
@@ -266,17 +462,18 @@ export const groupByDay = <T extends CloudJiraUser | ServerJiraUser>(worklogs: W
 };
 
 export const isAllowedDate = <T extends CloudJiraUser | ServerJiraUser>({
-    worklog, startDate, endDate, jiraUser, dates
+    worklog, startDate, endDate, dates
 }: WorklogAllowedDateParams<T>) => {
     const started = dayjs(worklog.started);
-    const isWorklogCreator = isCloudJiraUser(worklog.author) && isCloudJiraUser(jiraUser)
-        ? worklog.author.accountId === jiraUser.accountId
-        : (worklog.author as Worklog<ServerJiraUser>['author']).key === (jiraUser as ServerJiraUser).key;
 
-    if (!dates) return isWorklogCreator
-        && started.isAfter(startDate)
-        && started.isBefore(endDate);
+    if (!dates) return started.isAfter(startDate) && started.isBefore(endDate);
 
-    return isWorklogCreator
-        && dates.some(date => date.isSame(started, 'day'));
+    return dates.some(date => date.isSame(started, 'day'));
 };
+
+export const isWorklogCreator = <T extends CloudJiraUser | ServerJiraUser>({
+    worklog, jiraUser
+}: WorklogIsCreatorParams<T>) =>
+        isCloudJiraUser(worklog.author) && isCloudJiraUser(jiraUser)
+            ? worklog.author.accountId === jiraUser.accountId
+            : (worklog.author as Worklog<ServerJiraUser>['author']).key === (jiraUser as ServerJiraUser).key;
```

Comment:
why there's a need for assertion? Can modifying the declaration in params remove the need for assertion? If no then replace assertion with typeguards (e.g. if (typeof worklog.author) .... else throw Error/return false)